dali/python/nvidia/dali/plugin/pytorch/experimental/proxy/__init__.py

```diff
@@ -0,0 +1,517 @@
+#     |               |                  |                |                 |             |
+#     |               |           ---------------         |                 |             |
+#     |               |           | run         |         |                 |             |
+#     |               |           ---------------         |                 |             |
+#     |               |                  |                |                 |             |
+#     |               |                  |                |<~~put(req_0_1)~~|             |
+#     |               |                  |                |                 |             |
+#     |               |                  |                |-----------------|             |
+#     |               |                  |                |                 |             |
+#     |               |<~~put(data_1_0)~~|                |                 |             |
+#     |               |                  |                |                 |             |
+#     |               |----------------->|                |                 |             |
+#     |               |                  |                |                 |             |
+#     |<--data_1_0----|                  |                |                 |             |
+#     |               |                  |                |                 |             |
+# +-------+   +---------------+   +-------------+ +---------------+   +-----------+ +-----------+
+# | main  |   | dali_output_q |   | data_thread | | dali_input_q  |   | worker_0  | | worker_1  |
+# +-------+   +---------------+   +-------------+ +---------------+   +-----------+ +-----------+
+
+
+def _external_source_node_names(pipeline):
+    """
+    extract the names of all the ExternalSource nodes in the pipeline
+    """
+    # TODO(janton): Add a native function to query those names, so that we can do it
+    # also on deserialized pipelines
+    if pipeline._deserialized:
+        raise RuntimeError(
+            "Not able to find the external source "
+            "operator names, since the pipeline was deserialized"
+        )
+    if not pipeline._py_graph_built:
+        pipeline._build_graph()
+    input_node_names = []
+    for op in pipeline._ops:
+        if isinstance(op._op, ExternalSource):
+            input_node_names.append(op.name)
+    return input_node_names
+
+
+class DALIPipelineRunRef:
+    """
+    Reference for a DALI pipeline run iteration.
+    """
+
+    def __init__(self, batch_id, inputs):
+        """
+        batch_id: Identifier of the batch
+        is_scheduled: Whether the iteration has been scheduled for execution already
+        inputs: Inputs to be used when scheduling the iteration (makes sense only if
+        is_scheduled is False)
+        """
+        self.batch_id = batch_id
+        self.inputs = inputs
+        assert self.inputs is not None
+        self.is_scheduled = False
+
+
+class DALIProcessedSampleRef:
```

Comment:
Same here, should we consider providing a `__str__`/`__repr__` or some other information?